src/react-apps/applications/shared/src/navigation/NavMenu.tsx

```diff
@@ -11,26 +11,22 @@ import * as React from 'react';
 import {leftNavMenuSettings} from './leftNavMenuSettings';
 import {styles} from './navMenuStyles';
 
-export interface INavMenuState {
-  open: boolean;
-  openSubMenus: number[];
+export interface INavMenuProps {
+  drawerOpen: boolean;
+  classes: any;
+  handleToggleDrawer?: () => void;
 }
 
 export interface INavMenuProps extends WithStyles<typeof styles> {}
 
-class NavMenu extends React.Component<any, any> {
+class NavMenu extends React.Component<INavMenuProps, any> {
   public state = {
-    open: false,
     openSubMenus: [] as number[],
     selectedMenuItem: '',
   };
 
   public toggleNavMenu = () => {
-    this.setState((state: any) => {
-      return {
-        open: !state.open,
-      };
-    });
+    this.props.handleToggleDrawer();
```

Comment:
Optional prop should have a check if it is `null` or undefined.

Reply:
made this prop required.